Essentials/src/com/earth2me/essentials/Worth.java

```diff
@@ -62,6 +62,86 @@ public BigDecimal getPrice(IEssentials ess, ItemStack itemStack) {
         return result;
     }
 
+    /**
+     * Get the value of an item stack from the config for buying.
+     *
+     * @param ess       The Essentials instance.
+     * @param itemStack The item stack to look up in the config.
+     * @return The price from the config.
+     */
+    public BigDecimal getBuyPrice(IEssentials ess, ItemStack itemStack) {
+        BigDecimal result;
+
+        String itemname = itemStack.getType().toString().toLowerCase(Locale.ENGLISH).replace("_", "");
+
+        // Check for matches with data value from stack
+        // Note that we always default to BigDecimal.ONE.negate(), equivalent to -1
+        result = config.getBigDecimal("buy." + itemname + "." + itemStack.getDurability(), BigDecimal.ONE.negate());
+
+        // Check for matches with data value 0
+        if (result.signum() < 0) {
+            final ConfigurationSection itemNameMatch = config.getConfigurationSection("buy." + itemname);
+            if (itemNameMatch != null && itemNameMatch.getKeys(false).size() == 1) {
+                result = config.getBigDecimal("buy." + itemname + ".0", BigDecimal.ONE.negate());
```

Comment:
BigDecimal.ONE.negate() is used _so_ much... maybe makes sense to extract it away as a constant?

Reply:
I refactored the function and accounted for this change.